src/settings/project/mod.rs

```diff
@@ -0,0 +1,275 @@
+            }
+            ProjectType::Webpack => {
+                wranglerjs::run_build(self)?;
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn worker(&self) -> Result<Worker, failure::Error> {
+        self.build()?;
+        let worker = match self.project_type {
+            ProjectType::Rust => self.wasm_worker()?,
+            ProjectType::Webpack => self.webpack_worker()?,
+            ProjectType::JavaScript => self.js_worker()?,
+        };
+        // add other resoureces
+        Ok(worker)
+    }
+
+    fn wasm_worker(&self) -> Result<Worker, failure::Error> {
+        let name = krate::Krate::new("./")?.name.replace("-", "_");
+
+        build_generated_dir()?;
+        concat_js(&name)?;
+
+        let script_path = "./worker/generated/script.js";
+        let wasm_path = format!("./pkg/{}_bg.wasm", name);
+
+        let wasm_resource = wasm_resource(wasm_path)?;
+
+        Ok(Worker {
+            name: self.name.clone(),
+            script: Script {
+                name: "script".to_string(),
+                path: script_path.to_string(),
+            },
+            resources: vec![wasm_resource],
+        })
+    }
+
+    fn webpack_worker(&self) -> Result<Worker, failure::Error> {
+        let script_path = "./worker/script.js";
+        let mut worker = Worker {
+            name: self.name.clone(),
+            script: Script {
+                name: "script".to_string(),
+                path: script_path.to_string(),
+            },
+            resources: Vec::new(),
+        };
+        let wasm_path = "./worker/module.wasm";
+        if Path::new(wasm_path).exists() {
+            let wasm_resource = wasm_resource(wasm_path.to_string())?;
+            worker.resources.push(wasm_resource);
+        };
+        Ok(worker)
+    }
+
+    fn js_worker(&self) -> Result<Worker, failure::Error> {
+        let pkg = Package::new("./")?;
+        let script_path = pkg.main()?;
+        Ok(Worker {
+            name: self.name.clone(),
+            script: Script {
+                name: "script".to_string(),
+                path: script_path,
+            },
+            resources: Vec::new(),
+        })
+    }
+}
+
+fn wasm_resource(wasm_path: String) -> Result<Resource, failure::Error> {
+    let metadata_path = "./worker/metadata_wasm.json";
+    let mut binding = "wasm".to_string();
+
+    if Path::new(metadata_path).exists() {
+        let file = fs::File::open(metadata_path)?;
+        let reader = BufReader::new(file);
+        let data: serde_json::Value = serde_json::from_reader(reader)?;
+        binding = match data["bindings"][0]["name"].as_str() {
+            Some(s) => s.to_string(),
+            None => {
+                failure::bail!("binding was not a string");
+            }
+        };
+    }
+    Ok(Resource::WasmModule(WasmModule {
+        path: wasm_path.to_string(),
+        binding: binding,
+    }))
+}
+
+pub fn get_project_config() -> Result<Project, failure::Error> {
```

Comment:
this no longer needs to be a public function; users of this crate should call `Project::new` instead.